### PR TITLE
[Internal] SQL DOM : Refactor SQL DOM 

### PR DESCRIPTION
**Pull request by bchong95**
# Internal SQL DOM : Minor updates 

## Description

* touched up the names
* remove literal array collection
* Fix Identifier Path expression bug in to string
* made SqlQuery sealed

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Linq/ExpressionToSQL.cs [hunk 1]
```diff
@@ -468,16 +468,16 @@ private static SqlScalarExpression VisitBinary(BinaryExpression inputExpression,
...
             }
-            else if (right.Kind == SqlObjectKind.MemberIndexerScalarExpression && left.Kind == SqlObjectKind.LiteralScalarExpression)
+            else if (right is SqlMemberIndexerScalarExpression && left is SqlLiteralScalarExpression sqlLiteralScalarExpression)
             {
-                left = ExpressionToSql.ApplyCustomConverters(inputExpression.Right, left as SqlLiteralScalarExpression);
+                left = ExpressionToSql.ApplyCustomConverters(inputExpression.Right, sqlLiteralScalarExpression);
             }
 
-            return SqlBinaryScalarExpression.Create(op, left, right);
+            return SqlBinaryScalarExpression.Create(left, op, right);
```
Comment (sboshra) [inline Microsoft.Azure.Cosmos/src/Linq/ExpressionToSQL.cs:480]: please switch it back

File: Microsoft.Azure.Cosmos/src/SqlObjects/SqlBetweenScalarExpression.cs [hunk 2]
```diff
@@ -7,44 +7,44 @@ internal sealed class SqlBetweenScalarExpression : SqlScalarExpression
...
+            bool not,
+            SqlScalarExpression startInclusive,
+            SqlScalarExpression endExclusive)
             : base(SqlObjectKind.BetweenScalarExpression)
         {
             this.Expression = expression;
-            this.LeftExpression = leftExpression;
-            this.RightExpression = rightExpression;
-            this.IsNot = isNot;
+            this.Not = not;
```
Comment (sboshra) [inline Microsoft.Azure.Cosmos/src/SqlObjects/SqlBetweenScalarExpression.cs:15]: move to the end

File: Microsoft.Azure.Cosmos/src/SqlObjects/SqlBetweenScalarExpression.cs [hunk 3]
```diff
@@ -7,44 +7,44 @@ internal sealed class SqlBetweenScalarExpression : SqlScalarExpression
     {
         private SqlBetweenScalarExpression(
             SqlScalarExpression expression,
-            SqlScalarExpression leftExpression,
-            SqlScalarExpression rightExpression,
-            bool isNot = false)
+            bool not,
+            SqlScalarExpression startInclusive,
+            SqlScalarExpression endExclusive)
```
Comment (sboshra) [inline Microsoft.Azure.Cosmos/src/SqlObjects/SqlBetweenScalarExpression.cs:12]: endInclusive

File: Microsoft.Azure.Cosmos/src/SqlObjects/SqlInScalarExpression.cs [hunk 4]
```diff
@@ -8,60 +8,60 @@ namespace Microsoft.Azure.Cosmos.Sql
 
     internal sealed class SqlInScalarExpression : SqlScalarExpression
     {
-        private SqlInScalarExpression(SqlScalarExpression expression, bool not, IReadOnlyList<SqlScalarExpression> items)
+        private SqlInScalarExpression(SqlScalarExpression needle, bool not, IReadOnlyList<SqlScalarExpression> haystack)
```
Comment (sboshra) [inline Microsoft.Azure.Cosmos/src/SqlObjects/SqlInScalarExpression.cs]: consider reverting this

File: Microsoft.Azure.Cosmos/src/SqlObjects/SqlInScalarExpression.cs [hunk 5]
```diff
@@ -8,60 +8,60 @@ namespace Microsoft.Azure.Cosmos.Sql
...
         }
 
-        public bool Not
+        public SqlScalarExpression Needle
         {
             get;
         }
 
-        public SqlScalarExpression Expression
+        public bool Not
```
Comment (sboshra) [inline Microsoft.Azure.Cosmos/src/SqlObjects/SqlInScalarExpression.cs]: move to the end

File: Microsoft.Azure.Cosmos/src/Linq/ExpressionToSQL.cs (hunk 1 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Linq/ExpressionToSQL.cs:480]: done

File: Microsoft.Azure.Cosmos/src/SqlObjects/SqlInScalarExpression.cs (hunk 4 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/SqlObjects/SqlInScalarExpression.cs]: I like the needle and haystack approach.

File: Microsoft.Azure.Cosmos/src/SqlObjects/SqlInScalarExpression.cs (hunk 5 above)
Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/SqlObjects/SqlInScalarExpression.cs]: we can't put it after the bool since it's a required field now.

**Pull request merged**